syntax = "proto3";

service RsyncNotifications {
  rpc TakeSnapshotAfterRsyncCompletion (RsyncCompletionRequest) returns (SnapshotCompletionResponse);
  rpc PrepareDatasetBeforeRsyncStart (RsyncStartRequest) returns (PreparationCompletionResponse);
  rpc ListSnapshots (ListSnapshotsRequest) returns (ListSnapshotsResponse);
  rpc RecoverSnapshot (RecoverSnapshotRequest) returns (RecoverSnapshotResponse);
}

message RsyncCompletionRequest {
  string folder_name = 1;
}

message SnapshotCompletionResponse {
  bool success = 1;
  string message = 2;
}

message RsyncStartRequest {
  string folder_name = 1;
}

message PreparationCompletionResponse {
  bool success = 1;
  string message = 2;
}

message ListSnapshotsRequest {
  string folder_name = 1; // Optional argument
}

message ListSnapshotsResponse {
  bool success = 1;
  string message = 2;
  repeated string snapshots = 3;
}

message RecoverSnapshotRequest {
  string snapshot_name = 1;
  RecoveryMode mode = 2;
}

message RecoverSnapshotResponse {
  bool success = 1;
  string message = 2;
}

enum RecoveryMode {
  ORIGINAL = 0;
  STANDARD = 1;
}